### PR TITLE
Support Python 3

### LINKED ISSUE #77: Python 3 support
I get this error installing with pip3:

```
  Running setup.py install for xypath
      File "/usr/local/lib/python3.5/dist-packages/xypath/xypath.py", line 696
        raise AttributeError, "Bag has no attribute {!r}".format(name)
                            ^
    SyntaxError: invalid syntax
```

### PR DESCRIPTION
**Pull request by scraperdragon**
Surprisingly straightforward, revealed a hidden bug in xyzzy -- currently not fixed.

Fixes #77.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ENV HOME=/home/nobody \
+    PATH=/home/nobody/.local/bin:$PATH \
+    LANG=en_GB.UTF-8
+# LANG needed for httpretty install on Py3
+WORKDIR /home/nobody
+RUN pip install --user https://github.com/okfn/messytables/archive/py23.zip
+RUN pip3 install --user https://github.com/okfn/messytables/archive/py23.zip
+RUN pip install --user pyhamcrest
+RUN pip3 install --user pyhamcrest
+COPY . /home/nobody/
```
Comment (pwaller) [inline Dockerfile]: Don't be afraid of spacing to group things logically!

Also I'd recommend collapsing as many of the RUN's as you can.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+run:    build
+	@docker run \
+	    --rm \
+		-ti \
+	    xypath
```
Comment (pwaller) [inline Makefile]: Something a bit funny with the spacing here. Also, in scripts it is a great idea to use long form options (--terminal, --interactive, I think in this case), if they're available.

**Comment by StevenMaude** [comment]
- [ ] Tidy Dockerfile (as noted above)
- [ ] Investigate additional skipped tests
- [ ] Test on Python 2/3
- [ ] Merge and push a release to PyPI

**Comment by StevenMaude** [comment]
This is currently failing because of https://github.com/okfn/messytables/pull/157, though there's a fix I've submitted. If they don't merge soon, I'll fork it and add the fix there, though 9fd022c will need reviewing too.

File: Dockerfile [hunk 3]
```diff
@@ -20,5 +20,6 @@ ENV HOME=/home/nobody \
 WORKDIR /home/nobody
 RUN pip install --user https://github.com/okfn/messytables/archive/py23.zip
```
Comment (StevenMaude) [inline Dockerfile]: This URL is 404.

File: test/tcore.py [hunk 4]
```diff
@@ -51,7 +51,7 @@ def assertRaisesWithMessage(self, func, exception_type, msg, *args, **kwargs):
             self.fail('No exception was raised')
         except Exception as inst:
             self.assertIsInstance(inst, exception_type)
-            self.assertEqual(inst.message, msg)
+            self.assertEqual(inst.args[0], msg)
```
Comment (StevenMaude) [inline test/tcore.py]: Check if this can be nicer.

File: Dockerfile [hunk 5]
```diff
@@ -18,8 +18,6 @@ ENV HOME=/home/nobody \
     LANG=en_GB.UTF-8
 # LANG needed for httpretty install on Py3
 WORKDIR /home/nobody
-RUN pip install --user https://github.com/okfn/messytables/archive/py23.zip
```
Comment (StevenMaude) [inline Dockerfile]: So the 404 is fixed here, and the commit can be reduced.

File: test/tcore.py (hunk 4 above)
Comment (StevenMaude) [inline test/tcore.py]: Done.

File: Dockerfile (hunk 5 above)
Comment (StevenMaude) [inline Dockerfile]: Done.

File: Dockerfile (hunk 3 above)
Comment (StevenMaude) [inline Dockerfile]: (See below; done.)

**Comment by StevenMaude** [comment]
Finally done with this now, merging.

**Pull request merged**